torch/autograd/profiler_util.py

```diff
@@ -469,6 +518,8 @@ def __init__(
         self.thread: int = thread
         self.fwd_thread: Optional[int] = fwd_thread
         self.kernels: List[Kernel] = []
+        self.xpu_kernels: List[Kernel] = []
+        self.privateuse1_kernels: List[Kernel] = []
```

Comment:
What is the benefit of having 3 separate kernel lists (cuda, xpu, privateuse1)? Is there a scenario when two lists would co-exist? Also, I believe CUDA and AMD both share self.kernels.

Reply:
Thanks, I am okay with keeping separate kernel lists since the original solution of `use_device` is not very elegant in itself. Let's move forward with this solution, and maybe we can do a refactor in the future to improve this part.